include/ignition/math/Helpers.hh

```diff
@@ -830,6 +831,131 @@ namespace ignition
+                                                  // 01 - 19:
+                                                  // 20 - 23:
+
+          "([0-9]:|[0-5][0-9]:)){0,1}"            // minute:
+                                                  // 0 - 9:
+                                                  // 00 - 59:
+
+          "(?:([0-9]|[0-5][0-9]){0,1}"            // second:
+                                                  // 0 - 9
+                                                  // 00 - 59
+
+          "(\\.[0-9]{1,3}){0,1})$");              // millisecond:
+                                                  // .0 - .9
+                                                  // .00 - .99
+                                                  // .000 - 0.999
+      std::smatch matches;
+
+      // `matches` should always be a size of 6 as there are 6 matching
+      // groups in the regex.
+      // 1. The whole regex
+      // 2. The days
+      // 3. The hours
+      // 4. The minutes
+      // 5. The seconds
+      // 6. The milliseconds
+      // We can also index them as such below.
+      // Note that the space will remain in the day match, the colon
+      // will remain in the hour and minute matches, and the period will
+      // remain in the millisecond match
+      if (!std::regex_search(_timeString, matches, time_regex) ||
+          matches.size() != 6)
+        return timePoint;
+
+      uint64_t numberDays = 0;
+      uint64_t numberHours = 0;
+      uint64_t numberMinutes = 0;
+      uint64_t numberSeconds = 0;
+      uint64_t numberMilliseconds = 0;
+      std::string dayString = matches[1];
+      std::string hourString = matches[2];
+      std::string minuteString = matches[3];
+      std::string secondString = matches[4];
+      std::string millisecondString = matches[5];
+
+      // Days are the only unbounded number, so check first to see if stoi
+      // runs successfully
+      if (!dayString.empty())
+      {
+        // Erase the space
+        dayString.erase(dayString.length() - 1);
+        try
+        {
+          numberDays = std::stoi(dayString);
+        }
+        catch (const std::out_of_range &oor)
+        {
+          return timePoint;
+        }
+      }
+
+      if (!hourString.empty())
+      {
+        // Erase the colon
+        hourString.erase(hourString.length() - 1);
+        numberHours = std::stoi(hourString);
+      }
+
+      if (!minuteString.empty())
+      {
+        // Erase the colon
+        minuteString.erase(minuteString.length() - 1);
+        numberMinutes = std::stoi(minuteString);
+      }
+
+      if (!secondString.empty())
+      {
+        numberSeconds = std::stoi(secondString);
+      }
+
+      if (!millisecondString.empty())
+      {
+        // Erase the period
+        millisecondString.erase(0, 1);
+
+        // Multiplier because "4" = 400 ms, "04" = 40 ms, and "004" = 4 ms
+        numberMilliseconds = std::stoi(millisecondString) *
```

Comment:
Windows warning:

`'=': conversion from 'double' to 'uint64_t', possible loss of data`

Reply:
Fixed in 0f1ad24e9c42a527166a409b4257e034e8215beb, waiting for CI to show the warning's gone